build/parseSpec.c

```diff
@@ -462,6 +430,22 @@ int readLine(rpmSpec spec, int strip)
     lineType = parseLineType(s);
     if (!lineType)
 	goto after_classification;
+
+    /* check ordering of the conditional */
+    if (lineType->isConditional &&
+	(spec->readStack->lastConditional->id & lineType->wrongPrecursors)) {
+	/* the conditional can't be after %endif or the 1. conditional */
+	if (spec->readStack->lastConditional->id == LINE_ENDIF)
+	    rpmlog(RPMLOG_ERR,_("%s:%d: %s with no %%if\n"),
+		ofi->fileName, ofi->lineNum, lineType->text);
+	else
+	/* wrong ordering of the last two conditionals */
+	    rpmlog(RPMLOG_ERR,_("%s:%d: %s after %s\n"),
```

Comment:
Both the if-block and else-block consist of multiple physical lines, do use { } blocks in these cases to make it clearer. Also the /* wrong ordering */ comment is on odd indentation level here.

Reply:
Speaking of comments, the comments in this if-else construct don't really serve any purpose. The messages make it clear enough IMO.

Reply:
Oh, and one more thing I just realized: most of our error messages are of the format "%s: line %d: <message>", lets be consistent with that in both these messages.

Reply:
Erm, actually... I guess the most common message format is just "line %d: <message>" but in a few places filename is also included. All wonderfully inconsistent, and fixing all that is beyond scope of this pr. So guess that doesn't matter.